components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSInjectHandler.java

```diff
@@ -120,6 +121,8 @@ public boolean invoke(Object object, String name) throws SynapseException {
             //setting transport headers
             axis2MsgCtx.setProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS,
                                     JMSUtils.getTransportHeaders(msg, axis2MsgCtx));
+            // set transaction property
+            axis2MsgCtx.setProperty(BaseConstants.TRANSACTION, msg.getStringProperty(BaseConstants.TRANSACTION));
```

Comment:
Does the property need to be a String? Can it be a boolean?

Reply:
changed to a boolean.